sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxGatewayStoreModel.java

```diff
@@ -85,6 +95,22 @@ public RxGatewayStoreModel(
         this.sessionContainer = sessionContainer;
     }
 
+    void setGatewayServiceConfigurationReader(GatewayServiceConfigurationReader gatewayServiceConfigurationReader) {
+        this.gatewayServiceConfigurationReader = gatewayServiceConfigurationReader;
+    }
+
+    public void setPartitionKeyRangeCache(RxPartitionKeyRangeCache partitionKeyRangeCache) {
```

Comment:
can the GatewayStoreModel work without `partitionKeyRangeCache`?

it seems `partitionKeyRangeCache` is required now and it always set, so in that case this should be pass in the constructor not using setter method.

Reply:
Very good observation. To fill the cache we need GatewayStoreModel object ready itself, and master partition requests will work without any caches as we don't need session token for them. That is the reason we cant pass cache to GatewayStoreModel constructor , and have to set them explicitly once we initialize them , which is after GatewayStoreModel  object creation.